lib/rules/sort-invocations.js

```diff
@@ -0,0 +1,345 @@
+      },
+    })
+  );
+
+  node.modifiers = node.modifiers.map((modifier) => {
+    const { hash, loc, params, path } = modifier;
+
+    return b.elementModifier(path, params, hash, {
+      start: {
+        column: loc.start.column,
+        line: loc.start.line - 1,
+      },
+      end: {
+        column: loc.end.column,
+        line: loc.end.line - 1,
+      },
+    });
+  });
+}
+
+function sortAttributes(attributes) {
+  return attributes.sort(compareAttributes).map((attribute) => {
+    const { name, value } = attribute;
+
+    switch (value.type) {
+      case 'ConcatStatement': {
+        // Bug in ember-template-recast@6.1.5 (it removes the single character before or after a MustacheStatement)
+        const parts = value.parts.map(cloneIfTextNode);
+
+        // eslint-disable-next-line unicorn/prefer-spread
+        return b.attr(name, b.concat(parts));
+      }
+
+      case 'TextNode': {
+        // Bug in ember-template-recast@6.1.5 (it removes values that are an empty string)
+        if (value.chars === '') {
+          const { start, end } = value.loc;
+
+          const isValueUndefined = start.line === end.line && start.column === end.column;
+
+          if (!isValueUndefined) {
+            return b.attr(name, b.mustache(b.string('')));
+          }
+        }
+
+        break;
+      }
+    }
+
+    return b.attr(name, value);
+  });
+}
+
+function sortHash(hash) {
+  return b.hash(hash.pairs.sort(compareHashPairs));
+}
+
+function sortModifiers(modifiers) {
+  return modifiers.sort(compareModifiers).map((modifier) => {
+    const { hash, params, path } = modifier;
+
+    return b.elementModifier(path, params, hash);
+  });
+}
+
+export default class SortInvocations extends Rule {
+  /**
+   * @returns {import('./types.js').VisitorReturnType<SortInvocations>}
+   */
+  visitor() {
+    return {
+      BlockStatement(node) {
+        const { hash, params, path } = node;
+
+        let index = getUnsortedHashPairIndex(hash);
+
+        if (index === -1) {
+          return;
+        }
+
+        if (this.mode === 'fix') {
+          node = b.block(path, params, sortHash(hash), b.blockItself());
+        } else {
+          this.log({
+            isFixable: true,
+            message: `\`${getHashPairName(hash.pairs[index])}\` must appear after \`${getHashPairName(hash.pairs[index + 1])}\``,
+            node,
+          });
+        }
+      },
+
+      ElementNode(node) {
+        const { attributes, modifiers } = node;
+        let isSorted = false;
+
+        let index = getUnsortedAttributeIndex(attributes);
+
+        if (index !== -1) {
+          if (this.mode === 'fix') {
+            node.attributes = sortAttributes(attributes);
+            isSorted = true;
+          } else {
+            this.log({
+              isFixable: true,
+              message: `\`${getAttributeName(attributes[index])}\` must appear after \`${getAttributeName(attributes[index + 1])}\``,
```

Comment:
Due to how `getUnsortedAttributeIndex()` works, `index + 1` is guaranteed to be at most `attributes.length - 1`.